### PR TITLE
Rename IngredientsToSimpleMixturePredictor to IngredientsToFormulationPredictor

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR

## Description 
This is a misnomer, it creates a formulation that may or may not be a simple mixture. Mark the predictor as deprecated and create a new one with identical behavior. Also clean up some doc language, where "simple mixture" was often erroneously used where "formulation" was meant

Corresponding backend change: https://github.com/CitrineInformatics/mithril/pull/700

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [ ] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [ ] I have bumped the version in setup.py

### CODE REVIEW

File: src/citrine/informatics/predictors/ingredients_to_simple_mixture_predictor.py [hunk 1]
```diff
@@ -11,7 +12,15 @@
 
 class IngredientsToSimpleMixturePredictor(
         Resource['IngredientsToSimpleMixturePredictor'], Predictor, AIResourceMetadata):
-    """[ALPHA] A predictor interface that constructs a simple mixture from ingredient quantities.
+    """[DEPRECATED] [ALPHA] Constructs a simple mixture from ingredient quantities.
```
Comment (latture) [inline src/citrine/informatics/predictors/ingredients_to_simple_mixture_predictor.py:15]:
```suggestion
    """[DEPRECATED] Constructs a simple mixture from ingredient quantities.
```

File: src/citrine/informatics/predictors/ingredients_to_simple_mixture_predictor.py [hunk 2]
```diff
@@ -50,6 +59,11 @@ def __init__(self,
                  id_to_quantity: Mapping[str, RealDescriptor],
                  labels: Mapping[str, Set[str]],
                  archived: bool = False):
+        warn("{this_class} has been renamed. Please use {replacement} instead"
+             .format(this_class=self.__class__.__name__,
+                     replacement="Ingredients To Formulation Predictor"),
```
Comment (latture) [inline src/citrine/informatics/predictors/ingredients_to_simple_mixture_predictor.py:64]:
```suggestion
                     replacement=IngredientsToFormulationPredictor.__name__),
```
But you'll need to replace line 7 with
```
from citrine.informatics.predictors import Predictor, IngredientsToFormulationPredictor
```

**Review by latture — APPROVED** [review]
Only a few very minor suggestions, but those aren't blocking.

It's a bit strange that we'll deserialize deprecated predictors into the new type. There's a very small chance that could cause issues with client code (I think only if they're checking `isinstance`) or if we change the new predictor in the future such that we can't deserialize the old types. That risk is pretty small though.

File: src/citrine/informatics/predictors/ingredients_to_simple_mixture_predictor.py (hunk 2 above)
Comment (bfolie) [reply src/citrine/informatics/predictors/ingredients_to_simple_mixture_predictor.py:64]: Importing `IngredientsToFormulationPredictor` causes a bunch of tests to fail. I don't know why

**Comment by bfolie** [comment]
> It's a bit strange that we'll deserialize deprecated predictors into the new type. There's a very small chance that could cause issues with client code (I think only if they're checking `isinstance`) or if we change the new predictor in the future such that we can't deserialize the old types. That risk is pretty small though.

If we change the predictor itself, I think we'd have to go through the same hoops that we'd go through if we hadn't done this name change. But I agree that any client code which explicitly checks for the `IngredientsToSimpleMixturePredictor` type will break.

It seems unlikely enough that I feel comfortable making use of the fact that we're still in pre-release to make the breaking change. I'll send an announcement to #ds-de-prod

**Review by latture — APPROVED** [review]

**Pull request merged**